v2 invocation updates (#9768)
fishtown-analytics/docs.getdbt.com: website/docs/reference/programmatic-invocations.md

For [safe parallel execution](/reference/dbt-commands#available-commands), you can use the [<Constant name="platform_cli" />](/docs/platform/dbt-cli-installation) or [<Constant name="studio_ide" />](/docs/platform/studio-ide/develop-in-studio), both of which do that additional work to manage concurrency (multiple processes) on your behalf.

<VersionBlock firstVersion="2.0">

In v2, invocations are serialized through thread-level locks, so multiple invocations can't run concurrently within the same process. (In v1, parallel execution was unsupported but there was no locking, so invocations could still run in multithreaded mode.) As in v1, you can still parallelize by using multiprocessing to run each invocation in a separate process.

</VersionBlock>

## `dbtRunnerResult`

Each command returns a `dbtRunnerResult` object with the following attributes:

- `success` (bool): Whether the command succeeded.
- `result`: When the command completes (successfully or with handled errors), it returns the command's result(s). The return type varies by command.
- `exception`: When the dbt invocation encounters an unhandled error and does not complete, the exception that was raised.

> - `catalog`(v2 only): The catalog that the command produces when you request catalog generation. 

Reviewer: ```suggestion
- `catalog` (v2 only): The catalog that the command produces when you request catalog generation. 
```

<VersionBlock firstVersion="2.0">

> The v2 engine is implemented in Rust, so `exception` no longer contains the exact Python exception object raised by dbt. Instead, the caught error message is forwarded under an exception type:

Reviewer: ```suggestion
In <Constant name="core" /> 2.0, which runs on the <Constant name="fusion_engine" />, `exception` no longer contains the exact Python exception object raised by dbt. Instead, the caught error message is forwarded under an exception type:
```

- If the invocation fails at a foreign function interface (FFI) boundary before the engine picks up the invocation, `exception` contains an unwrapped exception type, such as `ValueError` or `RuntimeError`.
- If the invocation fails inside the engine, `exception` is a `DbtRunnerError`.

> V2 also adds a top-level `catalog` attribute to `dbtRunnerResult` when catalog generation is requested.

Reviewer: ```suggestion
v2 also adds a top-level `catalog` attribute to `dbtRunnerResult` when catalog generation is requested.
```

- `result`: When the command completes (successfully or with handled errors), it returns the command's result(s). The return type varies by command.

> - `exception`: When the dbt invocation encounters an unhandled error and does not complete, the exception that was raised.

Reviewer 2: Can we add `- catalog:` here for v2 ?
